src/pipecat/transports/network/webrtc_connection.py

```diff
@@ -343,6 +347,9 @@ async def connect(self):
             video_input_track = self.video_input_track()
             if video_input_track:
                 await self.video_input_track().discard_old_frames()
+            screen_video_input_track = self.screen_video_input_track()
+            if screen_video_input_track:
+                await self.screen_video_input_track().discard_old_frames()
             self.ask_to_renegotiate()
```

Comment:
Since you are already making some changes here, I included this `self.ask_to_renegotiate()` because sometimes the WebRTC connection was established before the bot was ready for receiving video. In those cases, we could lose some video frames, and it would take a couple of seconds before we received a key frame to finally start displaying video. So, this prevents this issue.

All this explanation just to say: I believe we should only invoke this if we have a `video_input_track` or `screen_video_input_track`. For audio-only, we shouldn’t need to invoke it.